src/rebar_packages.erl

```diff
@@ -243,7 +243,8 @@ parse_checksum(Checksum) ->
 
 update_package(Name, RepoConfig=#{name := Repo}, State) ->
     ?MODULE:verify_table(State),
-    try r3_hex_repo:get_package(get_package_repo_config(RepoConfig), Name) of
+    PkgRepoConfig1 = get_package_repo_config(RepoConfig),
```

Comment:
Not entirely sure about this one, as mentioned in https://github.com/erlang/rebar3/pull/2412#issuecomment-718286019.

Reply:
I'd probably ping @starbelly on this

Reply:
Will do.

Reply:
name is still used by rebar3_hex. I believe what we do is split a repo name like `hexpm:foo` and assign `hexpm` to name and `foo` to repo_name. Though, it should work without doing this (i.e., just assigning the full name to `repo_name`) but I will have to double check that. There are other keys used in rebar3 and rebar3_hex that are not used by hex_core, `parent` as an example. We could probably get away from this too, but I'm not sure there is an urgency to do this. I assume this is an issue here because dialyzer is complaining. Could we ignore it for the moment?

Reply:
We could ignore it, sure. But **why** would it be complaining though? Is it bad spec somewhere else? I can try and find a new cause and deal with it, if you think this new code would break.

Reply:
Let me check **where** it's coming from.

Reply:
`parent`, as-is, is not an issue, because even if used, is external to `rebar3`. Only `name` is important here.

Reply:
`rebar3` is calling vendored `r3_hex_repo` with a key that is not defined in `r3_hex_core`'s `config`. If that map contained `_ => _`, the issue would go away.
Since it doesn't, I'm assuming it's for good reason.

But I don't quite understand the flow of data, since I don't see how `rebar3_hex` is involved. Is the map somehow returned or used in a callback? (or transformed into some kind of term and used in an external request?)

Reply:
The only mention I find of `repo_name` in `rebar3_hex` is in a test.

Reply:
I asked in https://github.com/hexpm/hex_core/pull/105#issuecomment-723245312 if there's a possible solution on that end, since I already have a pull request opened there.

Reply:
> Since it doesn't, I'm assuming it's for good reason.

I wouldn't do that. It is probably my code :)

Reply:
> I wouldn't do that. It is probably my code :)

😄, so maybe you can help me figure out if the typespec is incomplete 😎

Reply:
> But I don't quite understand the flow of data, since I don't see how `rebar3_hex` is involved. Is the map somehow returned or used in a callback? (or transformed into some kind of term and used in an external request?)

Yes, repos found in an app's `rebar.config`(s) and from auth_config (`~/.config/rebar3/hex.config`) get put into rebar state. Repos are used when downloading packages from hex in rebar3 and when publishing packages and interacting with the hex api in general in rebar3_hex. 

Also, I see you're already removing `name` and so forth, looks like that should solve the issue coming from hex_core, maybe you mean the typespec for this function (`update_package/3`) ?

Reply:
@starbelly: aaf6583

Reply:
Yup, that looks fine to me 👍